packages/financial-templates-lib/src/price-feed/CoinGeckoPriceFeed.ts

```diff
@@ -53,17 +53,17 @@ class CoinGeckoPriceFeed extends PriceFeedInterface {
     };
   }
 
-  async update() {
-    const currentTime = this.getTime();
+  public async update(): Promise<void> {
+    const currentTime = await this.getTime();
 
     // Return early if the last call was too recent.
-    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+    if (this.lastUpdateTime !== null && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
```

Comment:
Yeah, I totally hear you. My goal is to _at least_ make the values that are initialized internally always be null rather than undefined. The codebase has a lot of overchecking because we don't know the types of things. Part of the advantage of typescript is to eliminate this by guaranteeing that the types have a fixed domain. I wanted to add some consistency to the way we build classes so that all internally managed values are initialized to null.

As for values that are passed to functions and constructors, you're right that this is a little more tricky because we don't know if the caller is using typescript. In this case, though, we're safe since this type is fully managed from within the class and typescript will protect against us assigning it to undefined.

Reply:
yea sounds good, that makes sense.